src/NServiceBus.Core/Pipeline/Incoming/TransportReceiveToPhysicalMessageProcessingConnector.cs

```diff
@@ -76,9 +75,10 @@ static void ConvertToPendingOperations(OutboxMessage deduplicationEntry, Pending
             }
         }
 
-        static List<TransportOperation> ConvertToOutboxOperations(IEnumerable<Transports.TransportOperation> operations)
+        static TransportOperation[] ConvertToOutboxOperations(Transports.TransportOperation[] operations)
```

Comment:
Took another look at this and realized that there are two different `TransportOperation` classes here, so now I understand what's going on. That's really confusing! When I first looked at this I though they were the same class coming in and going out and that made no sense.